File: apps/backend/src/main/java/com/example/dating/backend/profile/ProfileResponse.java
package com.example.dating.backend.profile;

import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.List;
import lombok.Builder;

@Builder
public record ProfileResponse(
    Long telegramId,
    String displayName,
    String bio,
    String gender,
    LocalDate birthday,
    String city,
    boolean vip,
    OffsetDateTime vipUntil,
    LocationPayload location,
    List<String> interests
) {

  @Builder
  public record LocationPayload(Double latitude, Double longitude, OffsetDateTime updatedAt) {}
}

File: apps/miniapp-gateway/src/modules/profiles/types/profile.response.ts
export interface ProfileLocation {
  latitude: number
  longitude: number
  updatedAt: string
}

export interface ProfileResponse {
  telegramId: number
  displayName: string
  bio: string | null
  gender: string | null
  birthday: string | null
  city: string | null
  vip: boolean
  vipUntil: string | null
  location: ProfileLocation | null
  interests: string[]
}
